azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -221,7 +221,7 @@ def _update(self, force_update):
         # Track goal state comes after that, the extensions will need the new certificate. The Agent needs to refresh the goal state in that
         # case, to ensure it fetches the new certificate.
         #
-        if self.extensions_goal_state.source == GoalStateSource.FastTrack:
+        if self._extensions_goal_state.source == GoalStateSource.FastTrack:
```

Comment:
Using the private member (_extensions_goal_state) instead of the public property (extensions_goal_state) for consistency with the rest of the code.